datasquare_test.go

```diff
@@ -420,6 +422,54 @@ func BenchmarkEDSRoots(b *testing.B) {
 	}
 }
 
+func BenchmarkEDSRootsWithErasuredNMT(b *testing.B) {
+	ODSSizeByteUpperBound := 512 * 1024 * 1024 // converting 512 MiB to bytes
+	totalNumberOfShares := float64(ODSSizeByteUpperBound) / shareSize
+	// the closest power of 2 of the square root of
+	// the total number of shares
+	nearestPowerOf2ODSSize := math.Pow(2, math.Ceil(math.Log2(math.Sqrt(
+		totalNumberOfShares))))
+	namespaceIDSize := 29
+
+	for i := 32; i <= int(nearestPowerOf2ODSSize); i *= 2 {
+		// number of shares in the original data square's row/column
+		odsSize := i
+		// number of shares in the extended data square's row/column
+		edsSize := 2 * odsSize
+		// generate an EDS with edsSize X edsSize dimensions in terms of shares.
+		// the generated EDS does not conform to celestia-app specs in terms
+		// of namespace version, also no erasure encoding takes place
+		// yet none of these should impact the benchmarking
+		ds := genRandSortedDS(edsSize, shareSize, namespaceIDSize)
+
+		// a tree constructor for erasured nmt
+		treeConstructor := newErasuredNamespacedMerkleTreeConstructor(uint64(i*2),
+			nmt.NamespaceIDSize(namespaceIDSize), nmt.IgnoreMaxNamespace(true),
+			nmt.InitialCapacity(i*2))
+
+		square, err := newDataSquare(ds, treeConstructor, shareSize)
+		if err != nil {
+			b.Errorf("Failure to create square of size %d: %s", i, err)
+		}
+		// the total size of the ODS in MiB
+		odsSizeMiBytes := i * i * shareSize / (1024 * 1024)
```

Comment:
[optional] why use `i` here?

```suggestion
		odsSizeMiBytes := squareSize * squareSize * shareSize / mebibyte
```

Reply:
Doen [a67997e](https://github.com/celestiaorg/rsmt2d/pull/304/commits/a67997ec08f9079f99271fd78dbb84af4bea74f2)